fenix/docs/architectureexample/HistoryNavigationMiddleware.kt

```diff
@@ -0,0 +1,27 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This is example code for the 'Simplified Example' section of
+// /docs/architecture-overview.md
+class HistoryNavigationMiddleware(
+    private val navController: NavController,
+) : Middleware<HistoryState, HistoryAction> {
+    override fun invoke(
+        context: MiddlewareContext<HistoryState, HistoryAction>,
+        next: (HistoryAction) -> Unit,
+        action: HistoryAction,
+    ) {
+        when(action) {
+            is HistoryAction.OpenItem -> {
+                next(action)
```

Comment:
Seems like we can extract `next(action)` above the when, similar to `TelemetryMiddleware` ? 
Same point for `StorageMiddleware`.

Reply:
Yeah, added some comments to explain the thinking there as well. Thanks!